requests-for-comments/2025-03-14-dw-testing-production-data.md

```diff
@@ -0,0 +1,89 @@
+# RFC - Test Pipeline Changes on Real Customer Data w/o Affecting Actual Prod Data
+
+## Background
+We now have reasonable traction with our data warehouse product and the cost of making breaking changes with
+incremental updates is growing as we accrue more and more external sources, self-managed sources, and saved queries each
+day. At this time, we don't have enough confidence in our (mostly unit) test suite and development workloads don't
+closely mimic the realities of production scenarios. Not all the sources have similar distributions of use, but one
+common theme is that many are SQL based and our data eventually end up in s3 via delta tables and are queryable via
+HogQl, so that feels like the area to hone our focus on for biggest impact. Other data sources like HubSpot, Salesforce,
+and Chargebee are less generic and also would have a smaller blast radius in the event syncing incremental or full table
+syncing gets regressed. These sources are still important and should be incorporated into the test plan as it matures.
+
+## Higher Level Goals
```

Comment:
What's the specific goal that would be great to solve here? Could you add specific issues we would want to protect against/detect with live testing (e.g. credentials failing because something with our live key for a source went wrong/expired)

Reply:
I will split out the higher level goals and testing goals into different subsections.